### DISCUSSION TITLE
Why can't numbers and English be rotated 180 degrees? Is there any way to solve this problem

### DISCUSSION THREAD

**Discussion by 858882847** [eclipse-birt/birt#1253] [Q&A]
I quote the syntax of CSS rotation plus font color. Only the font color has changed, and the numbers or English have not been rotated

**Comment by 858882847** [comment]
![V~5`H@8M6`3FHHX~R1SRI1H](https://user-images.githubusercontent.com/117696273/228999847-dad048f8-ef2a-4fca-853f-e91c2b6f604d.png)
"How can I achieve such an effect in numbers or English? I have tried all the methods without any effect."

**Reply by wimjongman** [reply]
please explain all the methods that you have tried in detail.

**Reply by 858882847** [reply]
Hello wimjongman, I quoted a css style and added font rotation. Writing a style in an html file has been tested on a web page and can achieve results. However, I wrote a css file and introduced it into BIRT. Only the color of digital fonts changes, and rotation has no effect. Writing 'this. getStyle(). textTransform='rotate (- 90deg)' in the BIRT onRender script has not taken effect. I don't know if there are other ways to achieve this.

**Reply by 858882847** [reply]
If there are examples, please send a script Thank you

**Comment by hvbtup** [comment]
BIRT does not fully support HTML and CSS (and never will). You can use the RotatedTextItem example.
It is also floating around in the internet as `birt.controls.lib_2.5.2.1.jar`. This is needed to use *at runtime* (so you have to append it to the ClassPath). At design time, you need the plugin with a similar name in addition. Add both *.jar files to BIRT's plugin folder.

**Reply by 858882847** [reply]
[test.zip](https://github.com/eclipse/birt/files/11122707/test.zip)
Hello, HVBTUP. Can you take a screenshot or give an example? I didn't understand the steps you said. Thank you

**Reply by hvbtup** [reply]
@SteveSchafer-Innovent For those developers not familiar with *building* Eclipse plugins, can you add binaries to the release on https://github.com/innoventsolutions/birt-controls-lib ?

**Reply by 858882847** [reply]
![UO8)@ (Y}CPG)XC@ZCI)6(1](https://user-images.githubusercontent.com/117696273/229662761-25b66a58-77d8-427a-bf86-370c6c88ff43.png)
I packaged these two projects into the BIRT plugin file, but the rotated text did not come out

**Reply by hvbtup** [reply]
Of course you also have to _use_ the RotatedText item in your report. You should find it in the palette in the BIRT designer.
Usage is straightforward: You specifiy the text to display and the rotation angle.

**Reply by 858882847** [reply]
You misunderstood what I meant. I put this plugin in the BIRT plugin and it didn't take effect. I also couldn't find the color palette for rotating text

**Comment by hvbtup** [comment]
It's part of the BIRT sources.
![grafik](https://user-images.githubusercontent.com/7644776/229440740-26f0e8f5-fd5e-4e88-a7bf-78932318a129.png)

@wimjongman Although the Roated Text Item has some quirks, I think it would make sense to include this plugin in the build artifacts for  BIRT 4.14, so that it can be used by BIRT report developers who are not necessarily Java experts.
Can you do whatever is necessary to achieve this?
